src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -311,7 +311,7 @@ private Task ProcessSendCmpctPayload(NetworkPeer node, SendCmpctPayload sendCmpc
                     ChainedBlock chainedBlock = blocksToAnnounce.Last();
                     if (chainedBlock != null)
                     {
-                        if (chainBehavior.PendingTip.GetAncestor(chainedBlock.Height) == null)
+                        if ((chainBehavior.PendingTip == null) || (chainBehavior.PendingTip.GetAncestor(chainedBlock.Height) == null))
```

Comment:
so now if I think about this in context of PendingTip can be null, then you need to make sure that all its uses are handled as such and you don't do that, so you should do it everywhere